core/src/main/java/org/apache/calcite/sql/dialect/MysqlSqlDialect.java

```diff
@@ -208,12 +212,52 @@ public boolean supportsAliasedValues() {
 
       unparseFloor(writer, call);
       break;
-
     default:
+      if (call.getOperator() == SqlStdOperatorTable.CONCAT) {
```

Comment:
Please move this piece of code before `switch-case` block, you can refer to the style in `OracleSqlDialect`.